common-service/src/main/java/org/sunbird/scheduler/SchedulerManager.java

```diff
@@ -0,0 +1,144 @@
+/** */
+package org.sunbird.scheduler;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Properties;
+import org.apache.commons.lang3.StringUtils;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.impl.StdSchedulerFactory;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.LoggerEnum;
+import org.sunbird.common.models.util.ProjectLogger;
+
+/**
+ * This class will manage all the Quartz scheduler. We need to call the schedule method at one time.
+ *
+ * @author Manzarul
+ */
+public abstract class SchedulerManager {
+
+  private static final String FILE = "quartz.properties";
+  public Scheduler scheduler = null;
+  public static SchedulerManager schedulerManager = null;
+
+  public SchedulerManager() {
+    initScheduler();
+  }
+
+  private void initScheduler() {
+    try {
+      Thread.sleep(240000);
+      boolean isEmbedded = false;
+      Properties configProp = null;
+      String embeddVal = System.getenv(JsonKey.SUNBIRD_QUARTZ_MODE);
+      if (JsonKey.EMBEDDED.equalsIgnoreCase(embeddVal)) {
+        isEmbedded = true;
+      } else {
+        configProp = setUpClusterMode();
+      }
+      if (!isEmbedded && configProp != null) {
+        ProjectLogger.log(
+            "SchedulerManager:initScheduler: Quartz scheduler is running in cluster mode.");
+        scheduler = new StdSchedulerFactory(configProp).getScheduler();
+      } else {
+        ProjectLogger.log(
+            "SchedulerManager:initScheduler: Quartz scheduler is running in embedded mode.");
+        scheduler = new StdSchedulerFactory().getScheduler();
+      }
+    } catch (Exception e) {
+      ProjectLogger.log("SchedulerManager:initScheduler: Error in starting scheduler jobs.", e);
+    } finally {
+      registerShutDownHook();
+    }
+    ProjectLogger.log("SchedulerManager:initScheduler: started scheduler jobs.");
+  }
+
+  /** This method will register the quartz scheduler job. */
+  public abstract void schedule();
+
+  /**
+   * Method to read the system properties and load to the Properties file that will be used to setup
```

Comment:
Creates a properties object containing PostgreSQL configuration (host, port, db, user, password).

PostgreSQL properties
IOException Exception in reading PostgreSQL configuration.